MODULE.bazel

```diff
@@ -1,12 +1,12 @@
 module(
     name = "gazelle",
-    version = "0.34.0",
+    version = "0.35.0",
     repo_name = "bazel_gazelle",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
 bazel_dep(name = "protobuf", version = "3.19.6", repo_name = "com_google_protobuf")
-bazel_dep(name = "rules_go", version = "0.42.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "rules_go", version = "0.44.0", repo_name = "io_bazel_rules_go")
```

Comment:
If tests pass without the update, we should not force it.

Reply:
It does appear to fail without the update. Not entirely sure the root cause: https://buildkite.com/bazel/bazel-gazelle/builds/3360#_

Reply:
Not sure either, probably the change in which I made Gazelle aware of `bazel_dep`s that are also Go modules requires a change in the rules_go setup. That's fine.